great_expectations/validator/validator.py

```diff
@@ -360,6 +377,304 @@ def inst_expectation(*args, **kwargs):
 
         return inst_expectation
 
+    def _build_expectation_configuration(
+        self,
+        expectation_type: str,
+        expectation_kwargs: dict,
+        meta: dict,
+        expectation_impl: "Expectation",  # noqa: F821
+    ) -> ExpectationConfiguration:
+        success_keys: Tuple[str] = (
+            expectation_impl.success_keys
+            if hasattr(expectation_impl, "success_keys")
+            else tuple()
+        )
+        arg_keys: Tuple[str] = (
+            expectation_impl.arg_keys
+            if hasattr(expectation_impl, "arg_keys")
+            else tuple()
+        )
+        runtime_keys: Tuple[str] = (
+            expectation_impl.runtime_keys
+            if hasattr(expectation_impl, "runtime_keys")
+            else None
+        ) or tuple()
+        # noinspection PyTypeChecker
+        override_keys: Tuple[str] = success_keys + arg_keys + runtime_keys
+
+        key: str
+        value: Any
+        expectation_kwargs_overrides: dict = {
+            key: value
+            for key, value in expectation_kwargs.items()
+            if key in override_keys
+        }
+
+        auto: Optional[bool] = expectation_kwargs.get("auto")
+        profiler_config: Optional[
+            Union[
+                RuleBasedProfilerConfig,
+                Dict[
+                    str,
+                    Union[
+                        str,
+                        float,
+                        Optional[Dict[str, Any]],
+                        Optional[Dict[str, Dict[str, Any]]],
+                    ],
+                ],
+            ]
+        ] = expectation_kwargs.get("profiler_config")
+        default_profiler_config: Optional[
+            Union[
+                RuleBasedProfilerConfig,
+                Dict[
```

Comment:
No Dict please, if at all possible. This doubles the code paths and more than doubles the defects

Reply:
@donaldheppner The dictionary is offered here under the assumption that the user can supply either a dictionary or an instantiated typed object.  I do not think that at this point we should require users to go an extra step and instantiate the object -- we have not held this stance thus far, because of maximal convenience we want to offer in our main user-facing API -- `validator.expect_*` (for other APIs, I agree that we should consider more canonical ways).  Thanks.

Reply:
See my suggestion above about isolating the logic merging different types of user input into a separate class and separate private methods so that all other methods and classes can use the typed ProfilerConfig object. This can be similar to the reconciliation methods on the RuleBasedProfiler.

Reply:
@donaldheppner This got done, based on our conversation.  Thanks!